### DISCUSSION TITLE
Disabling all queries on a refresh schedule

### DISCUSSION THREAD

**Discussion by edisonlowjy** [getredash/redash#6375] [Q&A]
Hi all,

Is there a way for me to turn off all queries that are on a scheduled refresh? We've written many queries over the course of years (about 1200) and most are no longer in use and I would like to start us off on a clean slate. Thanks

**Comment by justinclift** [comment]
No easy way for doing that is really springing to mind. :frowning:

Depending upon how familiar you are with scripting and/or programming, you *could* potentially write some kind of script that runs through all of your queries, disabling the ones that match what you want.

It'd probably take a few hours work, but *might* be worth the effort. :smile:

**Reply by edisonlowjy** [reply]
Thanks Justin! I am not very familiar with either scripting or programming, but would be keen to try and work it out with the help of ChatGPT. Do you have any thoughts off the top of your head how I can approach this?

**Reply by justinclift** [reply]
Oh.  Um... definitely make a backup of your entire Redash database, and your matching `/opt/redash/env` file first.

I've heard a mix of good and "pretty bad" outcomes from anything ChatGPT suggests.  So, you might get lucky.  But have a backup, just in case you don't.

With a backup, at least it'll *probably* be reasonably safe to try.  Unless ChatGPT writes code that tells your data sources to delete or modify their data.

Hmmm...

Actually, make sure you have full backups of anything Redash connects too first as well, just to be extra-paranoid.  (not joking) :smile:

---

For pointers to actually do it, I'd look at either of:

1. https://github.com/getredash/redash-toolbelt  <-- Python based utility functions for working with Redash
2. https://github.com/RecoLabs/redash-go-sdk  <-- Go based SDK for working with Redash

It's pretty likely both would have the functions you need, so either could do the job.

Note that I haven't done much with those myself, so that's just "in theory".

**Reply by edisonlowjy** [reply]
Thanks for that, I think it might be a bit too risky to attempt in that case, but appreciate your help anyways!

**Reply by justinclift** [reply]
Good call. :smile:
